src/components/HeaderMenuItems.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import { makeStyles, withStyles } from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+import HelpIcon from '@material-ui/icons/Help';
+import LockIcon from '@material-ui/icons/Lock';
+import MailIcon from '@material-ui/icons/Mail';
+import { LinkStyle } from '../styles/MermaidStyledComponents';
+
+import { mermaidHeader } from '../constants/header';
+
+const HeaderMenuItemsProperties = makeStyles(theme => ({
+  menuIconProperty: {
+    width: '20px',
+    height: '20px',
+    margin: '0 8px 2px 0'
+  }
+}));
+
+const StyledMenu = withStyles({
+  paper: {
+    border: '1px solid #d3d4d5',
+    marginLeft: '9px'
+  }
+})(props => (
+  <Menu
+    elevation={5}
+    getContentAnchorEl={null}
+    anchorOrigin={{
+      vertical: 'bottom'
+    }}
+    {...props}
+  />
+));
+
+const StyledMenuItem = withStyles(theme => ({
+  root: {
+    display: 'flex',
+    alignItems: 'center'
+  }
+}))(MenuItem);
+
+const HeaderMenuItems = ({ anchorEl, handleClose }) => {
+  const classes = HeaderMenuItemsProperties();
+  const iconFilter = name => {
```

Comment:
Another thought about your `iconFilter` function is that it _always_ returns a React component. You could make this function into a component itself:

```js
const MenuIcon = ({ name, menuIconProperty }) => {
  let Icon;

  switch (name) {
    case 'ABOUT THIS DATA':
      Icon = HelpIcon;
      break;
    case 'PRIVACY':
      Icon = LockIcon;
      break;
    case 'CONTACT':
      Icon = MailIcon;
      break;
    default:
      Icon = EditIcon;
      break;
  }
  return <Icon className={menuIconProperty} />;
};
```

And then when you wanted to call it, you could use the following:
```js
      <MenuIcon name={name} menuIconProperty={menuIconProperty} />
```

Reply:
(note: Because this a small thing, when I say that you can make it into a component, I'm not saying that it needs its own file. Just for clarification)